Modules/_multiprocessing/posixshmem.c

```diff
@@ -0,0 +1,724 @@
+    concludes. The caller is responsible for releasing the buffer.
+    */
+    int rc = 0;
+    NoneableName *p_name = (NoneableName *)checked_name;
+    PyObject *py_name_as_bytes = NULL;
+    char *p_name_as_c_string = NULL;
+
+    DPRINTF("inside convert_name_param\n");
+    DPRINTF("PyBytes_Check() = %d \n", PyBytes_Check(py_name_param));
+    DPRINTF("PyString_Check() = %d \n", PyString_Check(py_name_param));
+    DPRINTF("PyUnicode_Check() = %d \n", PyUnicode_Check(py_name_param));
+
+    p_name->is_none = 0;
+
+    // The name can be None or a Python string
+    if (py_name_param == Py_None) {
+        DPRINTF("name is None\n");
+        rc = 1;
+        p_name->is_none = 1;
+    }
+    else if (PyUnicode_Check(py_name_param) || PyBytes_Check(py_name_param)) {
+        DPRINTF("name is Unicode or bytes\n");
+        // The caller passed me a Unicode string or a byte array; I need a
+        // char *. Getting from one to the other takes a couple steps.
+
+        if (PyUnicode_Check(py_name_param)) {
+            DPRINTF("name is Unicode\n");
+            // PyUnicode_FSConverter() converts the Unicode object into a
+            // bytes or a bytearray object. (Why can't it be one or the other?)
+            PyUnicode_FSConverter(py_name_param, &py_name_as_bytes);
+        }
+        else {
+            DPRINTF("name is bytes\n");
+            // Make a copy of the name param.
+            py_name_as_bytes = PyBytes_FromObject(py_name_param);
+        }
+
+        // bytes_to_c_string() returns a pointer to the buffer.
+        p_name_as_c_string = bytes_to_c_string(py_name_as_bytes, 0);
+
+        // PyMalloc memory and copy the user-supplied name to it.
+        p_name->name = (char *)PyMem_Malloc(strlen(p_name_as_c_string) + 1);
+        if (p_name->name) {
+            rc = 1;
+            strcpy(p_name->name, p_name_as_c_string);
+        }
+        else
+            PyErr_SetString(PyExc_MemoryError, "Out of memory");
+
+        // The bytes version of the name isn't useful to me, and per the
+        // documentation for PyUnicode_FSConverter(), I am responsible for
+        // releasing it when I'm done.
+        release_bytes(py_name_as_bytes);
+    }
+    else
+        PyErr_SetString(PyExc_TypeError, "Name must be None or a string");
+
+    return rc;
+}
+
+
+
+/*   =====  Begin Shared Memory implementation functions ===== */
+
+static PyObject *
+shm_str(SharedMemory *self) {
+    return PyUnicode_FromString(self->name ? self->name : "(no name)");
+}
+
+static PyObject *
+shm_repr(SharedMemory *self) {
+    char mode[32];
+
+    sprintf(mode, "0%o", (int)(self->mode));
+
+    return PyUnicode_FromFormat("_posixshmem.SharedMemory(\"%s\", mode=%s)",
+                                self->name, mode);
+}
+
+static PyObject *
+my_shm_unlink(const char *name) {
+    DPRINTF("unlinking shm name %s\n", name);
+    if (-1 == shm_unlink(name)) {
+        switch (errno) {
```

Comment:
Why not use PyErr_SetFromErrno (of one of the other variants)?

Reply:
Removed in GH-11816.